test/functional/test_framework/messages.py

```diff
@@ -231,8 +231,8 @@ def serialize(self, *, with_time=True):
         return r
 
     def __repr__(self):
-        return "CAddress(nServices=%i ip=%s port=%i)" % (self.nServices,
```

Comment:
Any reason to not allow this constructor?

Reply:
It seems like the common pattern to create CAddress objects is to call `CAddress()` and then manually change time, nServices, ip, and port from their default value, so I don't see any reason why not to add these as optional arguments to the constructor. I think a couple other primitives could have their constructor take optional arguments too.

Reply:
I agree it's a common pattern to, to give another example, for a CTransaction with nVersion 2 to be created by calling CTransaction() and then manually changing nVersion.

However, I think it's out of scope for this PR.